lib/cve_server/app.rb

```diff
@@ -22,6 +22,21 @@ class App < Sinatra::Base
       end
     end
 
+    get '/v1/cves/:cves' do |cves|
+      answer = []
+      elements = cves.split(',').uniq
+      bad_request if elements.empty? || elements.size > 10
+      elements.each do |e|
```

Comment:
This is very cool action, I just have somesuggestions to improve this method:

1. It could be nice if you select and normalize the values before passing then to the query:
 ```
   normalized_cves = elements.select {|e| e.match?(/^CVE-\d{4}-(0\d{3}|[1-9]\d{3,})$/) }.map(&:upcase)
 ```

2. You also could add a new method to the CVEServer::Cve class:
  ```
 def all(pattern={})
      super(pattern).collect {|entry| remove_id(entry) }
 end
```

3. Use 1 query to fetch all the entrie instead invoking multiple times the find method:
```
  answer = CVEServer::Cve.all(id: { '$in': cves} )
```

https://docs.mongodb.com/manual/reference/operator/query/in/

Reply:
Addressed in 3d7762863c5c1796e2cbb954a8540c52b196442f

Reply:
Thanks for addressing it